src/borg/fuse.py

```diff
@@ -606,9 +615,21 @@ def opendir(self, inode, ctx=None):
     def read(self, fh, offset, size):
         parts = []
         item = self.get_item(fh)
-        for id, s, csize in item.chunks:
+        last_pos = self._last_pos.get(fh)
+
+        if not last_pos:
```

Comment:
general remark: if you use .get() and want afterwards to test for the "not found" case, always use `is None`.

Just using `if` or `if not` can easily introduce bugs if there can be falsy values in the cache dict.

This of course assumes that `None` is not in the valid values stored in the cache dict, otherwise one needs to use a different singleton as default and in the `is defaultobject` test.

Reply:
Sorry, my knowledge of python is very limited, I asked my friend to help me write this fix. I suppose it would be easy for project maintainers to just rewrite whole fix to be consistent with the project.